### PR TITLE
Add `assignments/` folder skeleton

### PR DESCRIPTION
**Pull request by teodutu**
Due to https://github.com/open-education-hub/operating-systems/issues/102, I can't check if the website works correctly on my machine. If any of you can, please do so and tell me if I did anything stupid.

### CODE REVIEW

**Comment by razvand** [comment]
I'm thinking of the student workflow for solving and submitting assignments. With the automatic checker solution (a.k.a. `vmchecker-baruta-weisz-xyz`), we want students to have their own (private) fork of an assignment that they would work on.

For this to happen, we could update the automatic checker solution to make use of this entire repository.

Another solution is to have each assignment in a separate repository. I find this cleaner and easier for students. One student could otherwise clone the entire repository and push the assignment solution (by mistake) in a public repository. For this other solution, there are two sub-options:

1. Use submodules and integrate them in this repository.
1. Have a workflow action that pushes assignment changes in this repository to each individual (read-only) assignment repository. This is the solution I like best, though it's a bit complicated.

**Comment by teodutu** [comment]
>     1. Use submodules and integrate them in this repository.
> 
>     2. Have a workflow action that pushes assignment changes in this repository to each individual (read-only) assignment repository. This is the solution I like best, though it's a bit complicated.

I also prefer option 2. Option 1 will lead to many scattered repos with past assignments, which will become difficult to manage over time. There's a general trend towards _monorepos_ anyway, and option 2 fits this paradigm. In addition, option 2 will make it easier for us to share our assignments with other teachers/universities.

This being said, I can't add that workflow required by option 2 now, as I don't know enough details about the students' repos yet. I'd say we first try to merge this PR to have a general structure for the assignments and then add the workflow later. This will allow us to work on each assignment independently and not be blocked by this PR. Does it sound OK @razvand?

**Review by gabrielmocanu — APPROVED** [review]
Sounds good to me.
Also I tested it locally and works like a charm.

**Comment by teodutu** [comment]
Ping @razvand. If you're OK with this, we can merge it and then start working on the assignments individually.

File: assignments/wrapper-calls/README.md
Comment (razvand) [inline assignments/wrapper-calls/README.md:1]:
```suggestion
# Mini Libc
```

File: config.yaml
Comment (razvand) [inline config.yaml:224]:
```suggestion
        - Mini Libc:
```

File: config.yaml
Comment (razvand) [inline config.yaml:225]:
```suggestion
            location: assignments/mini-libc
```

File: config.yaml
Comment (razvand) [inline config.yaml:227]: Use `Memory Allocator`.

File: config.yaml
Comment (razvand) [inline config.yaml:228]: Use `memory-allocator`.

File: assignments/electric-fence/README.md
Comment (razvand) [inline assignments/electric-fence/README.md:1]: Use `Memory Allocator`.

**Review by razvand — CHANGES_REQUESTED** [review]
@teodutu, see my comments on updating the assignment names. After that, we can merge the PR.

**Review by razvand — CHANGES_REQUESTED** [review]
@teodutu, please squash the two commits into the first one.

**Pull request merged**
